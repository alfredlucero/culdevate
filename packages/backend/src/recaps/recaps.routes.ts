import { Router } from "express";
import RecapsController from "./recaps.controller";
import authMiddleware from "../middleware/auth.middleware";
import validationMiddleware from "../middleware/validation.middleware";
import { RecapSchema } from "./recaps.validation";

const RecapsRoutes = {
  path: "/recaps",
  router: Router(),

  initializeRoutes() {
    // POST /recaps
    this.router.post("/", authMiddleware, validationMiddleware(RecapSchema, "body"), RecapsController.createRecap);

    // PATCH /recaps/:recapId
    this.router.patch(
      "/:recapId",
      authMiddleware,
      validationMiddleware(RecapSchema, "body"),
      RecapsController.updateRecap,
    );

    return this.router;
  },
};

export default RecapsRoutes;
