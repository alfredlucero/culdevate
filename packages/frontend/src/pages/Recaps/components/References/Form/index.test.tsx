import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import ReferencesForm, { ReferencesFormProps } from "./index";
import { RecapReferences, RecapKind } from "../../../recaps.interface";
import { MAX_GENERAL_LENGTH, recapReferencesErrors as validationErrors } from "../../../recaps.schema";
import * as RecapsService from "../../../recaps.service";
import { RecapFields } from "../../../recaps.schema";

const initialRecap: RecapReferences = {
  title: "Andrew C. - Product Manager",
  kind: RecapKind.References,
  userId: "userId",
  _id: "referencesId",
  company: "Sandia National Laboratories",
  bulletPoints: [],
  phoneNumber: "555-555-5555",
  email: "ac@sandia.test",
};
const defaultProps: ReferencesFormProps = {
  initialRecap,
  onSaveSuccess: () => {},
  isShowing: true,
  onHide: () => {},
};

describe("<ReferencesForm />", () => {
  test("should render without error", () => {
    const { container } = render(
      <ReferencesForm {...defaultProps} testId="referencesFormTestId" className="extra-class-name" />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render pre-populated form inputs given initial recap", () => {
    const { container } = render(
      <ReferencesForm {...defaultProps} testId="referencesFormTestId" className="extra-class-name" />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should show empty form inputs and disabled state given no initial recap", () => {
    const { container } = render(
      <ReferencesForm
        {...defaultProps}
        initialRecap={null}
        testId="referencesFormTestId"
        className="extra-class-name"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should show title required error after blurring with no title", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesTitle), { target: { value: "" } });
    fireEvent.blur(getByLabelText(RecapFields.referencesTitle));

    const requiredError = await findByText(validationErrors.titleRequired);

    expect(requiredError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show title max length error after blurring with title beyond max length", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesTitle), {
      target: { value: "a".repeat(MAX_GENERAL_LENGTH + 1) },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesTitle));

    const maxLengthError = await findByText(validationErrors.titleMaxLength);

    expect(maxLengthError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show title without inline error after blurring with valid title", async () => {
    const { getByLabelText, findByText, queryByText } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesTitle), { target: { value: "" } });
    fireEvent.blur(getByLabelText(RecapFields.referencesTitle));

    const requiredError = await findByText(validationErrors.titleRequired);
    expect(requiredError).toBeVisible();

    fireEvent.change(getByLabelText(RecapFields.referencesTitle), { target: { value: "title" } });
    fireEvent.blur(getByLabelText(RecapFields.referencesTitle));

    expect(queryByText(validationErrors.titleRequired)).toBeNull();
  });

  test("should show company required error after blurring with no company", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesCompany), { target: { value: "" } });
    fireEvent.blur(getByLabelText(RecapFields.referencesCompany));

    const requiredError = await findByText(validationErrors.companyRequired);

    expect(requiredError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show company max length error after blurring with company beyond max length", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesCompany), {
      target: { value: "a".repeat(MAX_GENERAL_LENGTH + 1) },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesCompany));

    const maxLengthError = await findByText(validationErrors.companyMaxLength);

    expect(maxLengthError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show company without inline error after blurring with valid company", async () => {
    const { getByLabelText, findByText, queryByText } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesCompany), { target: { value: "" } });
    fireEvent.blur(getByLabelText(RecapFields.referencesCompany));

    const requiredError = await findByText(validationErrors.companyRequired);
    expect(requiredError).toBeVisible();

    fireEvent.change(getByLabelText(RecapFields.referencesCompany), { target: { value: "company" } });
    fireEvent.blur(getByLabelText(RecapFields.referencesCompany));

    expect(queryByText(validationErrors.companyRequired)).toBeNull();
  });

  test("should show email max length error after blurring with email beyond max length", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesEmail), {
      target: { value: "email@test.com" + "a".repeat(MAX_GENERAL_LENGTH) },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesEmail));

    const maxLengthError = await findByText(validationErrors.emailMaxLength);

    expect(maxLengthError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show email invalid syntax error after blurring with invalid email", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesEmail), {
      target: { value: "invalidemail" },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesEmail));

    const invalidEmailError = await findByText(validationErrors.emailInvalid);

    expect(invalidEmailError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show email without inline error given valid/empty email", async () => {
    const { getByLabelText, queryByText } = render(<ReferencesForm {...defaultProps} initialRecap={null} />);

    expect(queryByText(validationErrors.emailInvalid)).toBeNull();

    fireEvent.change(getByLabelText(RecapFields.referencesEmail), {
      target: { value: "validemail@test.com" },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesEmail));

    expect(queryByText(validationErrors.emailInvalid)).toBeNull();
  });

  test("should show phone number max length error after blurring with phone number beyond max length", async () => {
    const { getByLabelText, findByText, getByTestId } = render(
      <ReferencesForm {...defaultProps} initialRecap={null} />,
    );

    fireEvent.change(getByLabelText(RecapFields.referencesPhoneNumber), {
      target: { value: "a".repeat(MAX_GENERAL_LENGTH + 1) },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesPhoneNumber));

    const maxLengthError = await findByText(validationErrors.phoneNumberMaxLength);

    expect(maxLengthError).toBeVisible();
    expect(getByTestId("referencesSaveButton")).toBeDisabled();
  });

  test("should show phone number without inline error given valid/empty phone number", async () => {
    const { getByLabelText, queryByText } = render(<ReferencesForm {...defaultProps} initialRecap={null} />);

    expect(queryByText(validationErrors.phoneNumberMaxLength)).toBeNull();

    fireEvent.change(getByLabelText(RecapFields.referencesPhoneNumber), {
      target: { value: "555-555-5555" },
    });
    fireEvent.blur(getByLabelText(RecapFields.referencesPhoneNumber));

    expect(queryByText(validationErrors.phoneNumberMaxLength)).toBeNull();
  });

  test("should call onHide after clicking Cancel button", () => {
    const onHideMock = jest.fn();
    const { getByText } = render(<ReferencesForm {...defaultProps} initialRecap={null} onHide={onHideMock} />);

    fireEvent.click(getByText("Cancel"));

    expect(onHideMock).toHaveBeenCalled();
  });

  test("should call onSaveSuccess after saving successfully with valid fields", async () => {
    const onSaveSuccessMock = jest.fn();
    const { getByTestId } = render(<ReferencesForm {...defaultProps} onSaveSuccess={onSaveSuccessMock} />);

    jest.spyOn(RecapsService, "updateRecap").mockImplementationOnce(() => Promise.resolve(initialRecap));

    expect(getByTestId("referencesSaveButton")).not.toBeDisabled();
    fireEvent.click(getByTestId("referencesSaveButton"));

    await waitFor(() => {
      expect(onSaveSuccessMock).toHaveBeenCalled();
    });
  });

  test("should not call onSaveSuccess after failing to save successfully with valid fields", async () => {
    const onSaveSuccessMock = jest.fn();
    const { getByTestId, findByText } = render(<ReferencesForm {...defaultProps} onSaveSuccess={onSaveSuccessMock} />);

    jest.spyOn(RecapsService, "updateRecap").mockImplementationOnce(() => Promise.reject({}));

    expect(getByTestId("referencesSaveButton")).not.toBeDisabled();
    fireEvent.click(getByTestId("referencesSaveButton"));

    await findByText("Something went wrong with updating a References Recap!", { exact: false });
    expect(onSaveSuccessMock).not.toHaveBeenCalled();
  });
});
