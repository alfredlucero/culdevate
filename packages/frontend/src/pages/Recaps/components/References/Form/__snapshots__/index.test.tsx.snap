// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferencesForm /> should render pre-populated form inputs given initial recap 1`] = `
<div>
  <div
    class="mt-4 extra-class-name"
    data-testid="referencesFormTestId"
  >
    <form>
      <div
        class="w-full mb-4"
      >
        <div
          class="mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="title"
          >
            Reference Name/Title
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="title"
            name="title"
            placeholder="What is the name or title of someone you worked with?"
            type="text"
            value="Andrew C. - Product Manager"
          />
        </div>
      </div>
      <div
        class="w-full mb-4"
      >
        <div
          class="mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="company"
          >
            Company
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="company"
            name="company"
            placeholder="Which company did you both work together at?"
            type="text"
            value="Sandia National Laboratories"
          />
        </div>
      </div>
      <div
        class="flex w-full mb-4"
      >
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="email"
          >
            Email
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="email"
            name="email"
            placeholder="What is your reference's email?"
            type="text"
            value="ac@sandia.test"
          />
        </div>
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="phoneNumber"
          >
            Phone Number
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="phoneNumber"
            name="phoneNumber"
            placeholder="What was your reference's phone number?"
            type="text"
            value="555-555-5555"
          />
        </div>
      </div>
      <div>
        <span
          class="block text-gray-700 text-sm font-bold mb-2"
        >
          Bullet Points
        </span>
        <div
          class="mb-6"
        >
          <div
            class="bg-gray-300 pt-6 px-4 pb-4 bullet-point-list-droppable"
            data-rbd-droppable-context-id="1"
            data-rbd-droppable-id="bulletPointInputList"
          >
            <div
              aria-labelledby="rbd-lift-instruction-1"
              class="bg-white hover:bg-blue-100 rounded"
              data-rbd-drag-handle-context-id="1"
              data-rbd-drag-handle-draggable-id="referencesBulletPoint-0"
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="referencesBulletPoint-0"
              draggable="false"
              tabindex="0"
            >
              <div
                class="flex items-start px-4 py-6 rounded shadow-md cursor-pointer mb-2"
              >
                <div
                  class="font-sans text-2xl mr-4 text-teal-300"
                  id=""
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-square fa-w-14 "
                    data-icon="square"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="mr-4 flex-1"
                >
                  
                  <input
                    aria-required="true"
                    class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
                    id="referencesBulletPoint-0"
                    name="referencesBulletPoint-0"
                    placeholder="Provide your highlights and results..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="font-sans text-3xl text-red-500 hover:text-red-700 cursor-pointer"
                  id="referencesBulletPoint-0"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-trash-alt fa-w-14 "
                    data-icon="trash-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mt-4 flex items-center"
                type="button"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="font-sans text-2xl mr-2 add-bullet-point-plus-icon"
                    id=""
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-plus fa-w-14 "
                      data-icon="plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  Add Bullet Point
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Cancel
          </div>
        </button>
        <button
          class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
          data-testid="referencesSaveButton"
          type="submit"
        >
          <div
            class="flex items-center"
          >
            Save Recap
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<ReferencesForm /> should render without error 1`] = `
<div>
  <div
    class="mt-4 extra-class-name"
    data-testid="referencesFormTestId"
  >
    <form>
      <div
        class="w-full mb-4"
      >
        <div
          class="mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="title"
          >
            Reference Name/Title
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="title"
            name="title"
            placeholder="What is the name or title of someone you worked with?"
            type="text"
            value="Andrew C. - Product Manager"
          />
        </div>
      </div>
      <div
        class="w-full mb-4"
      >
        <div
          class="mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="company"
          >
            Company
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="company"
            name="company"
            placeholder="Which company did you both work together at?"
            type="text"
            value="Sandia National Laboratories"
          />
        </div>
      </div>
      <div
        class="flex w-full mb-4"
      >
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="email"
          >
            Email
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="email"
            name="email"
            placeholder="What is your reference's email?"
            type="text"
            value="ac@sandia.test"
          />
        </div>
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="phoneNumber"
          >
            Phone Number
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="phoneNumber"
            name="phoneNumber"
            placeholder="What was your reference's phone number?"
            type="text"
            value="555-555-5555"
          />
        </div>
      </div>
      <div>
        <span
          class="block text-gray-700 text-sm font-bold mb-2"
        >
          Bullet Points
        </span>
        <div
          class="mb-6"
        >
          <div
            class="bg-gray-300 pt-6 px-4 pb-4 bullet-point-list-droppable"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="bulletPointInputList"
          >
            <div
              aria-labelledby="rbd-lift-instruction-0"
              class="bg-white hover:bg-blue-100 rounded"
              data-rbd-drag-handle-context-id="0"
              data-rbd-drag-handle-draggable-id="referencesBulletPoint-0"
              data-rbd-draggable-context-id="0"
              data-rbd-draggable-id="referencesBulletPoint-0"
              draggable="false"
              tabindex="0"
            >
              <div
                class="flex items-start px-4 py-6 rounded shadow-md cursor-pointer mb-2"
              >
                <div
                  class="font-sans text-2xl mr-4 text-teal-300"
                  id=""
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-square fa-w-14 "
                    data-icon="square"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="mr-4 flex-1"
                >
                  
                  <input
                    aria-required="true"
                    class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
                    id="referencesBulletPoint-0"
                    name="referencesBulletPoint-0"
                    placeholder="Provide your highlights and results..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="font-sans text-3xl text-red-500 hover:text-red-700 cursor-pointer"
                  id="referencesBulletPoint-0"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-trash-alt fa-w-14 "
                    data-icon="trash-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mt-4 flex items-center"
                type="button"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="font-sans text-2xl mr-2 add-bullet-point-plus-icon"
                    id=""
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-plus fa-w-14 "
                      data-icon="plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  Add Bullet Point
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Cancel
          </div>
        </button>
        <button
          class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
          data-testid="referencesSaveButton"
          type="submit"
        >
          <div
            class="flex items-center"
          >
            Save Recap
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<ReferencesForm /> should show empty form inputs and disabled state given no initial recap 1`] = `
<div>
  <div
    class="mt-4 extra-class-name"
    data-testid="referencesFormTestId"
  >
    <form>
      <div
        class="w-full mb-4"
      >
        <div
          class="mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="title"
          >
            Reference Name/Title
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="title"
            name="title"
            placeholder="What is the name or title of someone you worked with?"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="w-full mb-4"
      >
        <div
          class="mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="company"
          >
            Company
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="company"
            name="company"
            placeholder="Which company did you both work together at?"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="flex w-full mb-4"
      >
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="email"
          >
            Email
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="email"
            name="email"
            placeholder="What is your reference's email?"
            type="text"
            value=""
          />
        </div>
        <div
          class="w-1/2 mr-4"
        >
          <label
            class="block text-gray-700 text-sm font-bold mb-2"
            for="phoneNumber"
          >
            Phone Number
          </label>
          <input
            aria-required="true"
            class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
            id="phoneNumber"
            name="phoneNumber"
            placeholder="What was your reference's phone number?"
            type="text"
            value=""
          />
        </div>
      </div>
      <div>
        <span
          class="block text-gray-700 text-sm font-bold mb-2"
        >
          Bullet Points
        </span>
        <div
          class="mb-6"
        >
          <div
            class="bg-gray-300 pt-6 px-4 pb-4 bullet-point-list-droppable"
            data-rbd-droppable-context-id="2"
            data-rbd-droppable-id="bulletPointInputList"
          >
            <div
              aria-labelledby="rbd-lift-instruction-2"
              class="bg-white hover:bg-blue-100 rounded"
              data-rbd-drag-handle-context-id="2"
              data-rbd-drag-handle-draggable-id="referencesBulletPoint-0"
              data-rbd-draggable-context-id="2"
              data-rbd-draggable-id="referencesBulletPoint-0"
              draggable="false"
              tabindex="0"
            >
              <div
                class="flex items-start px-4 py-6 rounded shadow-md cursor-pointer mb-2"
              >
                <div
                  class="font-sans text-2xl mr-4 text-teal-300"
                  id=""
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-square fa-w-14 "
                    data-icon="square"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="mr-4 flex-1"
                >
                  
                  <input
                    aria-required="true"
                    class="shadow appearance-none border w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline rounded"
                    id="referencesBulletPoint-0"
                    name="referencesBulletPoint-0"
                    placeholder="Provide your highlights and results..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="font-sans text-3xl text-red-500 hover:text-red-700 cursor-pointer"
                  id="referencesBulletPoint-0"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-trash-alt fa-w-14 "
                    data-icon="trash-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="flex justify-center"
            >
              <button
                class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mt-4 flex items-center"
                type="button"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="font-sans text-2xl mr-2 add-bullet-point-plus-icon"
                    id=""
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-plus fa-w-14 "
                      data-icon="plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  Add Bullet Point
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Cancel
          </div>
        </button>
        <button
          class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 opacity-50 cursor-not-allowed "
          data-testid="referencesSaveButton"
          disabled=""
          type="submit"
        >
          <div
            class="flex items-center"
          >
            Save Recap
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;
